### ISSUE TITLE
Hash-map malfunction notice `Phel\..\Map\IndexedNode could not be converted to int`

### ISSUE THREAD

**Issue by jasalt** [phel-lang/phel-lang#786]
- Phel v0.16.0 dev-main (4002bec8e8a612545f4338d504ad25509c4a6310)
- PHP 8.2.25 (cli)

TL;DR creating hash-map with certain data triggers notice, see short example below posted as comment.

Original legthy report:

Handling a dataset of 10000 barcodes (mostly EAN13, some mixed formats) and noticed some odd behavior I think might partially have something to do with crc32 hashing function collisions when I create hash-map using barcodes as keys. Simplified & anonymized example data (some EANs scrambled with https://github.com/theonlypwner/crc32):

```
(def data
  [{:post_id 426 :meta_value ""}
   {:post_id 497 :meta_value "0Mlq3z"}
   {:post_id 537 :meta_value "bTOQLr"}
   {:post_id 546 :meta_value "P0dH3t"}
   {:post_id 548 :meta_value "1238924228357"}
   {:post_id 564 :meta_value "AB18924230123"}
   {:post_id 569 :meta_value "BC1892423005"}
   {:post_id 576 :meta_value "A7I8924230I8"}
   {:post_id 580 :meta_value "987182423015"}
   {:post_id 585 :meta_value "9871892423017"}
   {:post_id 609 :meta_value "JF0010"}
   {:post_id 614 :meta_value "JF1102"}
   {:post_id 622 :meta_value "JF1017"}
   {:post_id 624 :meta_value "JF1015"}
   {:post_id 632 :meta_value "JF1260"}
   {:post_id 636 :meta_value "JF1261"}
   {:post_id 648 :meta_value "jiKUic"}
   {:post_id 715 :meta_value "PU4NVO"}
   ])

(count data)
# -> 18

(def result
  (for [{:meta_value barcode :post_id pid} :in
        data :reduce [acc {}]]
    (put acc barcode pid)))

# Notice: Object of class Phel\Lang\Collections\Map\IndexedNode could not be converted to int in /path/to/my-app/vendor/phel-lang/phel-lang/src/php/Lang/Collections/Map/ArrayNode.php on line 53

(count result)
# -> 18
```

The notice comes from quite deep and does not make sense in context of my code. I can get rid of it by a couple means that I can't make much sense either, following code does not trigger notice:

```
# Setting hash-map value as string:
(for [{:meta_value barcode :post_id pid} :in
      data :reduce [acc {}]]
  (put acc barcode (str pid)))

# Skip empty barcodes:
(for [{:meta_value barcode :post_id pid} :in
      data :reduce [acc {}] :when (not (empty? barcode))]
  (put acc barcode pid))

```

Thinking if empty meta_values in data might lead to issue, but it does not seem to be the case:

```
(def data2
  [{:post_id 426 :meta_value ""}
   {:post_id 497 :meta_value ""}  # set this as empty string too
   {:post_id 537 :meta_value "bTOQLr"}
   {:post_id 546 :meta_value "P0dH3t"}
   {:post_id 548 :meta_value "1238924228357"}
   {:post_id 564 :meta_value "AB18924230123"}
   {:post_id 569 :meta_value "BC1892423005"}
   {:post_id 576 :meta_value "A7I8924230I8"}
   {:post_id 580 :meta_value "987182423015"}
   {:post_id 585 :meta_value "9871892423017"}
   {:post_id 609 :meta_value "JF0010"}
   {:post_id 614 :meta_value "JF1102"}
   {:post_id 622 :meta_value "JF1017"}
   {:post_id 624 :meta_value "JF1015"}
   {:post_id 632 :meta_value "JF1260"}
   {:post_id 636 :meta_value "JF1261"}
   {:post_id 648 :meta_value "jiKUic"}
   {:post_id 715 :meta_value "PU4NVO"}
   ])

(for [{:meta_value barcode :post_id pid} :in
        data2 :reduce [acc {}]]
  (put acc barcode pid))

# no notice triggered
```

Removing product maps from the original example data makes notice to disappear:

```
(def data3
  [{:post_id 426 :meta_value ""}
   {:post_id 497 :meta_value "0Mlq3z"}
   {:post_id 537 :meta_value "bTOQLr"}
   {:post_id 546 :meta_value "P0dH3t"}
   {:post_id 548 :meta_value "1238924228357"}
   {:post_id 564 :meta_value "AB18924230123"}
   {:post_id 569 :meta_value "BC1892423005"}
   {:post_id 576 :meta_value "A7I8924230I8"}
   {:post_id 580 :meta_value "987182423015"}
   {:post_id 585 :meta_value "9871892423017"}
   {:post_id 609 :meta_value "JF0010"}
   {:post_id 614 :meta_value "JF1102"}
   {:post_id 622 :meta_value "JF1017"}
   {:post_id 624 :meta_value "JF1015"}
   {:post_id 632 :meta_value "JF1260"}
   {:post_id 636 :meta_value "JF1261"}
   {:post_id 648 :meta_value "jiKUic"}
   # {:post_id 715 :meta_value "PU4NVO"}  # <- commented out
   ])

(for [{:meta_value barcode :post_id pid} :in
        data3 :reduce [acc {}]]
  (put acc barcode pid))
# no notice triggered
```

Lastly, noted that making slightest changes to letters in data cause the notice disappear, .

```
(def data4
  [{:post_id 426 :meta_value ""}
   {:post_id 497 :meta_value "0Mlq3zAAAA"}  # adding more that 3 A's here makes notice disappear
   {:post_id 537 :meta_value "bTOQLr"}
   {:post_id 546 :meta_value "P0dH3t"}
   {:post_id 548 :meta_value "1238924228357"}
   {:post_id 564 :meta_value "AB18924230123"}
   {:post_id 569 :meta_value "BC1892423005"}
   {:post_id 576 :meta_value "A7I8924230I8"}
   {:post_id 580 :meta_value "987182423015"}
   {:post_id 585 :meta_value "9871892423017"}
   {:post_id 609 :meta_value "JF0010"}
   {:post_id 614 :meta_value "JF1102"}
   {:post_id 622 :meta_value "JF1017"}
   {:post_id 624 :meta_value "JF1015"}
   {:post_id 632 :meta_value "JF1260"}
   {:post_id 636 :meta_value "JF1261"}
   {:post_id 648 :meta_value "jiKUic"}
   {:post_id 715 :meta_value "PU4NVO"}
   ])
(for [{:meta_value barcode :post_id pid} :in
      data4 :reduce [acc {}]]
  (put acc barcode pid))
# no notice
```

I was unable to anonymize some of these EAN codes without retaining the notice by other means than reversing the crc32 which caused me to think that maybe it's related to Phel [hasher functionality](https://github.com/phel-lang/phel-lang/blob/f89f425685ba79f4ad5a2fc10190417f6da0d0a7/src/php/Lang/Hasher.php#L70).

The expected behavior would be that the notice would not be triggered in any of the cases, or it should include more informative message.

**Comment by jasalt** [comment]
To simplify the issue, it seems that this hash-map is cursed:

```
{
 "" 426
 "0Mlq3zAAA" 497
 "bTOQLr" 537
 "P0dH3t" 546
 "1238924228357" 548
 "AB18924230123" 564
 "BC1892423005" 569
 "A7I8924230I8" 576
 "987182423015" 580
 "9871892423017" 585
 "JF0010" 609
 "JF1102" 614
 "JF1017" 622
 "JF1015" 624
 "JF1260" 632
 "JF1261" 636
 "jiKUic" 648
 "PU4NVO" 715
 }

## (out) PHP Notice:  Object of class Phel\Lang\Collections\Map\IndexedNode could not be converted to int in /home/user/dev/phel-async-amphp-demo/vendor/phel-lang/phel-lang/src/php/Lang/Collections/Map/ArrayNode.php on line 53

## (out) {9871892423017 585  426 PU4NVO 715 JF1260 632 JF1015 624 jiKUic 648 JF1017 622 AB18924230123 564 P0dH3t 546 bTOQLr 537 0Mlq3zAAA 497 JF0010 609 JF1102 614 A7I8924230I8 576 987182423015 580 1238924228357 548 JF1261 636 BC1892423005 569}
```

**Comment by jasalt** [comment]
@Chemaclass this could be unpinned from issues list also as it shows up there still while having been closed.
